### ISSUE TITLE
why the accuracy_reward stay 0.5 all the time?

### ISSUE THREAD

**Issue by Lynnzake** [FareedKhan-dev/train-deepseek-r1#4]
Here's part of the log 
```python
{'eval_loss': 0.0006179613410495222, 'eval_runtime': 284.3483, 'eval_samples_per_second': 0.348, 'eval_steps_per_second': 0.025, 'eval_rewards/accuracy_reward': 0.5, 'eval_rewards/format_reward': 0.0, 'eval_rewards/reasoning_steps_reward': 0.8933333548903465, 'eval_rewards/repetition_penalty_reward': -0.011207603123039008, 'eval_reward': 1.382125735282898, 'eval_reward_std': 0.1532403835374862, 'eval_completion_length': 251.8875, 'eval_kl': 0.015430915802717209, 'epoch': 0.01}
{'loss': 0.0007, 'grad_norm': 3.265625, 'learning_rate': 4.969630038652678e-06, 'rewards/accuracy_reward': 0.5, 'rewards/format_reward': 0.0, 'rewards/reasoning_steps_reward': 0.8979166895151138, 'rewards/repetition_penalty_reward': -0.01163857274223119, 'reward': 1.386278110742569, 'reward_std': 0.14473110824474134, 'completion_length': 251.5625, 'kl': 0.017456819815561176, 'epoch': 0.01}
{'loss': 0.0009, 'grad_norm': 3.65625, 'learning_rate': 5.107675317504142e-06, 'rewards/accuracy_reward': 0.5, 'rewards/format_reward': 0.0, 'rewards/reasoning_steps_reward': 0.9125000238418579, 'rewards/repetition_penalty_reward': -0.009585449565201997, 'reward': 1.4029145538806915, 'reward_std': 0.135728381248191, 'completion_length': 252.26875, 'kl': 0.02297338340431452, 'epoch': 0.01}
{'loss': 0.0012, 'grad_norm': 4.125, 'learning_rate': 5.245720596355605e-06, 'rewards/accuracy_reward': 0.5, 'rewards/format_reward': 0.0, 'rewards/reasoning_steps_reward': 0.922916692495346, 'rewards/repetition_penalty_reward': -0.00911087888525799, 'reward': 1.4138058125972748, 'reward_std': 0.1266389235795941, 'completion_length': 251.125, 'kl': 0.030348912812769413, 'epoch': 0.01}
{'loss': 0.0016, 'grad_norm': 3.984375, 'learning_rate': 5.383765875207068e-06, 'rewards/accuracy_reward': 0.5, 'rewards/format_reward': 0.0, 'rewards/reasoning_steps_reward': 0.9500000178813934, 'rewards/repetition_penalty_reward': -0.010074394347611815, 'reward': 1.439925640821457, 'reward_std': 0.09026941693155095, 'completion_length': 249.56875, 'kl': 0.03993788044899702, 'epoch': 0.01}
{'loss': 0.0017, 'grad_norm': 4.3125, 'learning_rate': 5.5218111540585315e-06, 'rewards/accuracy_reward': 0.5, 'rewards/format_reward': 0.0, 'rewards/reasoning_steps_reward': 0.9541666746139527, 'rewards/repetition_penalty_reward': -0.009727871022187173, 'reward': 1.4444387912750245, 'reward_std': 0.10285891313105822, 'completion_length': 239.20625, 'kl': 0.04340738980099559, 'epoch': 0.01}
```

I notice the accuracy reward stay the same all the time, but the log doesn't print `"Warning: Failed to parse gold solution:"`.

**Comment by FareedKhan-dev** [comment]
Yeah, it didn't catch the actual gold solution, same reason mentioned in this [issue](https://github.com/FareedKhan-dev/train-deepseek-r1/issues/2#issuecomment-2742930018).
